File: src/ProjectEuler/C/Problems/Problem5.c
// 232792560 - Completed 8.21.2013.
#include <stdlib.h>
#include <stdio.h>

#include "../C_Utils/number.h"

int main(){
    long f = 1;
    long t,n;
    for (n = 1;n <= 20;n++){
	t = lcm(f,n);
	f = t;
    }
    printf("%ld\n",t);
    return 0;
}

File: src/ProjectEuler/Problems/Problem5.hs
-- 232792560 - Completed 29.4.2013 - Learned about folds.
module ProjectEuler.Problems.Problem5
    ( problem5
    ) where

import Data.List (foldl)

problem5 :: IO ()
problem5 = print $ foldl (lcm) 1 [1..20]
